packages/vite/src/node/plugins/worker.ts

```diff
@@ -356,14 +356,19 @@ export function webWorkerPlugin(config: ResolvedConfig): Plugin {
           return result()
         }
       }
-      if (!isWorker) {
-        const workerMap = workerCache.get(config)!
-        workerMap.assets.forEach((asset) => {
-          this.emitFile(asset)
-          workerMap.assets.delete(asset.fileName!)
-        })
-      }
       return result()
+    },
+
+    generateBundle(opts) {
```

Comment:
This was moved because

- hooks run in `generateBundle` because `renderChunk(worker) -> renderChunk(legacy) -> generateBundle(worker) -> generateBundle(legacy)`
- `opts.__vite_skip_asset_emit__` is injected in `renderChunk(legacy)`

. Is my understanding correct?

Reply:
yes, `generateBundle` can got a correct  `opts.__vite_skip_asset_emit__` value. 👍